### DISCUSSION TITLE
Lung nodule detection multiclass training from scratch

### DISCUSSION THREAD

**Discussion by junxiant** [Project-MONAI/MONAI#7917] [Q&A] [answered]
Hello, I've been trying to re-train the detection mode from scratch, but instead of doing detection for just "nodule" class, I have modified the dataset_fold0.json file, specifically the "label", by using the labelling from LIDC-IDRI dataset, at pylidc https://pylidc.github.io/annotation.html . Basically matching the json file "image" value and pylidc seriesinstanceuid.

So now there are a total of 5 classes, 0,1,2,3,4 . 

I have modified the config file fg_labels to [0,1,2,3,4]. 

I have modified the luna16_training.py at [line 245 ](https://github.com/Project-MONAI/tutorials/blob/main/detection/luna16_training.py#L245 )

```
before
    coco_metric = COCOMetric(classes=["nodule"], iou_list=[0.1], max_detection=[100])
after
    coco_metric = COCOMetric(classes=["Non_Solid_GGO","Non_Solid_Mixed","Part_Solid_Mixed","Solid_Mixed","Solid"], iou_list=[0.1], max_detection=[100])
```

But after the second validation after epoch 10 the val metrics are showing all 0.
![image](https://github.com/user-attachments/assets/ac86179b-91c4-4b5f-a9a0-25d31b536666)

Is there something i need to modify to make this work for multiclass? Or would i need to wait for a few more validation to confirm that it is not training properly

Here are some tf logs graphs
![image](https://github.com/user-attachments/assets/2f2bc091-1d53-48dd-ac8c-9e2c32ee53da)

**Comment by KumoLiu** [comment]
Hi @Can-Zhao, can you help share some comments on this one? Thanks!

**Comment by Can-Zhao** [comment, marked as answer]
Thank you for reaching out.
- The coco metric may be very small in the first a few epochs. You may wait for 50 epochs to see. 
- You maybe add --verbose flag. Details about matched anchors during training will be printed out. You can see if the hyper-parameters you set is correct.
- Have you looked at the visualized image and box in the tensorboard? Or maybe you could try a small lr and see how it goes?

**Reply by junxiant** [reply]
> Thank you for reaching out.
> 
> * The coco metric may be very small in the first a few epochs. You may wait for 50 epochs to see.
> * You maybe add --verbose flag. Details about matched anchors during training will be printed out. You can see if the hyper-parameters you set is correct.
> * Have you looked at the visualized image and box in the tensorboard? Or maybe you could try a small lr and see how it goes?

Hello, u are right, it needed more rounds of validation. I was able to train it fully with the modified dataset labels, but it gave really poor results when i ran the evaluation script. i wonder if it makes sense to use the same evaluation script? since the original was for one "nodule" class but i have five classes now. mAR seems great, but mAP is very low

![image](https://github.com/user-attachments/assets/2649b213-ef67-4e5c-96e7-5642233d93fb)

**Reply by Can-Zhao** [reply]
The evaluation script provided by LUNA challenge was not designed for multi-class detection. I have never tested it for multi-class detection. I would feel reporting Coco metric should be enough. Or you may run the evaluation script for each class separately.

**Reply by junxiant** [reply]
yes the coco metrics should be enough. the evaluation script does not work for multi class.

thanks for the help~

**Reply by junxiant** [reply]
oh just one more question
1. are there any plans to integrate more detection architectures into monai?
